Zend/zend_hash.h

```diff
@@ -853,13 +853,18 @@ static zend_always_inline void *zend_hash_next_index_insert_mem(HashTable *ht, v
 {
 	zval tmp, *zv;
 
-	ZVAL_PTR(&tmp, NULL);
-	if ((zv = zend_hash_next_index_insert(ht, &tmp))) {
-		Z_PTR_P(zv) = pemalloc(size, GC_FLAGS(ht) & IS_ARRAY_PERSISTENT);
-		memcpy(Z_PTR_P(zv), pData, size);
-		return Z_PTR_P(zv);
+	void *p = pemalloc(size, GC_FLAGS(ht) & IS_ARRAY_PERSISTENT);
+	if (!p) {
+		return NULL;
 	}
-	return NULL;
+
+	memcpy(p, pData, size);
+	ZVAL_PTR(&tmp, p);
+	if (!(zv = zend_hash_next_index_insert(ht, &tmp))) {
```

Comment:
I wouldn't assign the return value but I'd simply return p at the end.

Reply:
thx simpler indeed.